### PR TITLE
docs: add documentation for tests

### LINKED ISSUE #620: Improve Test Documentation
**Is your feature request related to a problem? Please describe.**
Currently tests and their structure are up to each developer to understand. There's no documentation of how they should be organized, written, tips, etc.

**Describe the solution you'd like**
Write a `test.md` or `readme.md` file within the `tests/` directory documenting test usage, creation, etc.

**Additional context**
I intend to open a PR that resolves this issue, just creating this issue so this is being tracked.

### PR DESCRIPTION
**Pull request by treatybreaker**
Nitpick my writing skills, no holds barred. I'm not an exemplary writer, so if you see something you want changed, just say "change this to x" and I'll make the modification, no need to justify 😃.

Closes #620

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+### Where do utility functions go?
+
+If you have any utility code that has to do with git, it should be placed in `tests/util/git_harness.lua`.
+
+If you have a generic utility function _only_ relevant for tests then it should go in `tests/util/util.lua`. If it is generic enough that it could be useful in the general Neogit code then a consideration should be made to place this utility code in `lua/neogit/lib/util.lua`.
+
+### Where should raw content files go?
+
+Raw content files that you want to test against should go into `tests/test_files`. If you have a raw file you'd like to use against in git, then you'll need to add it to the git repository within `tests/.repo`. This can be done by changing directory into `tests/.repo` and renaming `.git.orig` to `.git` then adding any relevant changes to that repository. Once you're done, make sure you rename `.git` back to `.git.org`.
```
Comment (CKolkey) [inline tests/README.md:55]: I would prefer `tests/fixtures/` or `tests/supports/ to `test/test_files/` - what do you think?

Comment (treatybreaker) [reply tests/README.md:55]: Of the two I prefer 'fixtures'. Will refactor to that in a bit.

Comment (treatybreaker) [reply tests/README.md]: @CKolkey Done.

File: tests/README.md
Comment (ten3roberts) [inline tests/README.md:1]: Could we link this file or refer to it in some way from `CONTRIBUTING.md` now that it is merged?

**Review by ten3roberts — CHANGES_REQUESTED** [review]

File: tests/README.md
[the remark this answers is not in the document]
Comment (treatybreaker) [reply tests/README.md:1]: Sorry about the delay, my email never popped up for this review 4 days ago(?!).

Why don't we keep Contributing largely as it is and link to the test documentation from there as the tests document is a bit large to have as a single section within it.

Comment (treatybreaker) [reply tests/README.md:1]: Yes 🙂

**Comment by treatybreaker** [comment]
Updated with the changes requested. Again sorry about the **4** day delay, never saw the notification from my email.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ A [`Makefile`](./Makefile) is set up to run tests.
 
 ```sh make test ```
```
Comment (ten3roberts) [inline CONTRIBUTING.md:79]: We should make this a multiline fenced block and remove the odd space at the end.

Just a minor nitpick

Comment (ten3roberts) [reply CONTRIBUTING.md:79]: Oops, realized that was there already, must have been line squashed during some formatting. If you want to, feel free to fix them up and give them some breathing room

**Review by ten3roberts — APPROVED** [review]

**Comment by ten3roberts** [comment]
> Updated with the changes requested. Again sorry about the **4** day delay, never saw the notification from my email.

No worries. That happens to all of us. My mailbox has been absolutely flooding and then some more from all different corners of all different programming languages, domains, and chats.

I'm just glad you are offering to help, it is always such a lovely thing in open source.

File: CONTRIBUTING.md (hunk 2 above)
Comment (treatybreaker) [reply CONTRIBUTING.md:79]: Sure, I can clean that up. Not a problem 😄.

Comment (treatybreaker) [reply CONTRIBUTING.md]: Done.

**Review by ten3roberts — APPROVED** [review]
Looks great. We could merge this now if you want to.

If there is anything you discover that is missing in this feel free to open another PR, I figure it is better to merge this so that you don't have to many outstanding PRs generating merge conflicts.

**Comment by treatybreaker** [comment]
> Looks great. We could merge this now if you want to.
> 
> If there is anything you discover that is missing in this feel free to open another PR, I figure it is better to merge this so that you don't have to many outstanding PRs generating merge conflicts.

Yeah go ahead and merge, I have nothing to add currently and if I do I'll add it in another PR.

**Pull request merged**